tools/validators/instance_validator/error_guide.md

```diff
@@ -0,0 +1,1712 @@
+```
+
+## Onboarded Device with Missing Translation
+
+### Description:
+Devices that report to the cloud (containing a cloud device id) must have a defined translation, either the existing one or an updated one, unless the entity is being deleted.
+
+### Example Bad Config:
+```
+# entity with cloud_device_id specified but no translations; fails as default operation is ADD
+
+DMP_EDM-17-GUID:
+  type: HVAC/DMP_EDM
+  code: DMP_EDM-17
+  cloud_device_id: "1234567890123456"
+
+US-SEA-BLDG1-GUID:
+  type: FACILITIES/BUILDING
+  code: US-SEA-BLDG1
+```
+
+### Error Message:
+```
+[ERROR]	Entity (DMP_EDM-17-GUID: DMP_EDM-17) Has a type which has defined fields but this instance has neither links nor a translation.
+[ERROR]	Entity DMP_EDM-17-GUID (DMP_EDM-17) has a cloud_device_id but is missing a translation. Reporting devices must have a translation when cloud_device_id is present; unless the operation is DELETE
+```
+
+## No Provided Fields
+
+### Description:
+Non-facility entities must have defined fields or link to a reporting device with their fields defined.
+
+### Example Bad Config:
+```
+SDC_EXT-17-GUID:
+  type: HVAC/SDC_EXT
+  etag: a12345
+  code: SDC_EXT-17
+
+CONFIG_METADATA:
+  operation: UPDATE
+
+US-SEA-BLDG1-GUID:
+  type: FACILITIES/BUILDING
+  etag: b23456
+  code: US-SEA-BLDG1
+```
+
+### Error Message:
+```
+[ERROR]	Entity (SDC_EXT-17-GUID: SDC_EXT-17) Has a type which has defined fields but this instance has neither links nor a translation.
+```
+
+## Duplicate Key/GUID
+
+### Description:
+Each defined entity must have a unique GUID.
+
+### Example Bad Config:
+```
+SDC_EXT-17-GUID:
+  type: HVAC/SDC_EXT
+  code: SDC_EXT-17
+
+SDC_EXT-17-GUID:
+  type: HVAC/SDC_EXT
+  code: SDC_EXT-17
+
+US-SEA-BLDG1-GUID:
+  type: FACILITIES/BUILDING
+  code: US-SEA-BLDG1
+```
+
+### Error Message:
+```
+While parsing
+  in "<unicode string>", line 4, column 1:
+    SDC_EXT-17-GUID:
+    ^ (line: 4)
+Duplicate key 'SDC_EXT-17-GUID' found
+  in "<unicode string>", line 4, column 16:
+    SDC_EXT-17-GUID:
+                   ^ (line: 4)
+```
+
+## Repeated Types
+
+### Description:
+Each entity must have a single defined type.
+
+### Example Bad Config:
+```
+US-SEA-BLDG1-GUID:
+  type: FACILITIES/BUILDING
+  type: FACILITIES/123456b
+  code: US-SEA-BLDG1
+```
+
+### Error Message:
+```
+While parsing
+  in "<unicode string>", line 3, column 3:
+      type: FACILITIES/123456b
+      ^ (line: 3)
+Duplicate key 'type' found
+  in "<unicode string>", line 3, column 7:
+      type: FACILITIES/123456b
+          ^ (line: 3)
+```
+
+## Duplicate Metadata
+
+### Description:
+A config can only have one metadata block.
+
+### Example Bad Config:
+```
+SDC_EXT-18-GUID:
+  code: SDC_EXT-18
+
+CONFIG_METADATA:
+  operation: UPDATE
+
+CONFIG_METADATA:
+  operation: INITIALIZE
+
+US-SEA-BLDG1-GUID:
+  type: FACILITIES/BUILDING
+  code: US-SEA-BLDG1
+```
+
+### Error Message:
```

Comment:
This error message doesn't seem to match the issue of having two CONFIG_METADATA blocks defined